src/averageAges.js

```diff
@@ -1,61 +1,43 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filterPeople = century
+    ? people.filter(({ sex, died }) =>
+      (sex === 'm' && Math.ceil(died / 100) === century))
+    : people.filter(({ sex }) => sex === 'm');
+
+  const maleAverageAge = filterPeople.reduce((sum, { born, died }) => (
+    sum + died - born), 0) / filterPeople.length;
```

Comment:
Checklist point 2

Reply:
Also, you do the same on calculateWomenAverageAge. You can create separate function and you it here and on calculateWomenAverageAge